### ISSUE TITLE
[bug]: vae selection doesn't seen to be working, still using the models vae

### ISSUE THREAD

**Issue by Vargol** [invoke-ai/InvokeAI#4284]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

macOS

### GPU

mps

### What version did you experience this issue on?

3.0.2post1

### What happened?

I've been trying to use the fp16 bit vae  by madebyollin but was still getting black images if the vae was run with the
precision set to fp16.

<img width="453" alt="Screenshot 2023-08-14 at 14 40 14" src="https://github.com/invoke-ai/InvokeAI/assets/62868/cc15446c-141e-4561-8986-c64e8b98238b">

I checked the logs and found the following that suggested it was still using the models default vae

```
[2023-08-14 14:19:20,659]::[InvokeAI]::DEBUG --> Unloading model /Volumes/Sabrent Media/Documents/Source/Python/ai_models/invokeai/models/sdxl/main/sdxl_base_fp16:sdxl:main:unet to free 0.16 GB (-4.78 GB)
[2023-08-14 14:19:20,943]::[InvokeAI]::DEBUG --> Moving /Volumes/Sabrent Media/Documents/Source/Python/ai_models/invokeai/models/sdxl/main/sdxl_
base_fp16:sdxl:main:vae into mps
```

I then renamed be vae directory in the model and softlinked the fp16 fixed vae directory there instead and the images were no longer black but fully formed.

Poking around I saw this which looks like it might be related the parameters passed to render, with vae set to null

```
            "latents_to_image": {
                "id": "latents_to_image",
                "is_intermediate": false,
                "type": "l2i",
                "latents": null,
                "vae": null,
                "tiled": false,
                "fp32": false,
                "metadata": null
            },
```        

    

### Screenshots

https://user-images.githubusercontent.com/62868/260816997-cc15446c-141e-4561-8986-c64e8b98238b.png

**Comment by Vargol** [comment]
This appears to be fixed in the main branch, closing

**Comment by QuestionQuest117** [comment]
I'm still encountering this issue in main. What did you do to manually fix it exactly? That was unclear. Could you please provide more details?

**Comment by Vargol** [comment]
I found the vae in the invoke models cache

```
invokeai/models/sdxl/vae/sdxl-1-0-vae-fix
```

I then found the SDXL model in the cache too

```
invokeai/models/sdxl/main/sdxl_base_fp16
```
Note I have installed copy of the 16bit base model,
as when I originally installed SDXL it downloaded the full 32bit version and would sometimes take 4 or 5 minutes to load and convert to 16bit MPS. Thats been fixed in  3.0.2.
You probably need this directory instead

```
stable-diffusion-xl-base-1-0
```

in there is a vae directory

```
invokeai/models/sdxl/main/sdxl_base_fp16/vae
```

I renamed that vae directory
and softlinked the fixed vae directory (or you could copy it there instead)

```
mv vae vae_default
ln -s ../../vae/sdxl-1-0-vae-fix vae
```

So now when it uses the models default vae its actually using the fixed vae instead.
